plugins/history_plugin/history_plugin.cpp

```diff
@@ -205,7 +205,7 @@ namespace eosio {
          }
 
          void record_account_action( account_name n, const action_trace& act ) {
-            auto& chain = chain_plug->chain();
+            auto& chain = app().get_method<methods::get_controller>()();
```

Comment:
I would rather see this cached on the plugin instead of looking up the method each time as these are called very offen.